### PR TITLE
fix empty variables not shown in '\raw'

### LINKED ISSUE #257: empty variables not shown in `\raw`
Empty variables should show in `\raw`.

For example, currently `usql` produces the following output:

```sh
(not connected)=> \set a ''
(not connected)=> select :a
(not connected)-> \p
select 
(not connected)-> \raw
select 
(not connected)->
```

However, on the `\raw` above, it should have shown:

```sh
(not connected)-> \raw
select :a
(not connected)->
```

Comparatively:

```sh
(not connected)=> \set a b
(not connected)=> select :a
(not connected)-> \p
select b
(not connected)-> \raw
select :a
(not connected)->
```

And, it behaves as expected for quoted variables:

```sh
(not connected)=> \set a ''
(not connected)=> select :'a'
(not connected)-> \p
select ''
(not connected)-> \raw
select :'a'
(not connected)->
```

It's worth noting this is not dependent on the length of the variable name:

```sh
(not connected)=> select :aa
(not connected)-> \p
select 
(not connected)-> \raw
select 
(not connected)-> 
```

The reason for this behavior is partly due to the original way that variables were parsed and interpolated (at time of entry). This will require a slight overhaul to how variables are captured and interpolated.

### PR DESCRIPTION
**Pull request by xwjdsh**
Close #257

### CODE REVIEW

File: stmt/stmt.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ type Var struct {
 	Name string
 	// Len is the length of the replaced variable.
 	Len int
+	// Exist indicates whether the variable has been defined.
+	Exist bool
```
Comment (nineinchnick) [inline stmt/stmt.go:26]: `Exist` is a bit generic, how about `IsDefined`?
```suggestion
	IsDefined bool
```

Comment (kenshaw) [reply stmt/stmt.go:26]: `Defined` ...

File: stmt/stmt_test.go [hunk 2]
```diff
@@ -254,6 +254,29 @@ func TestNextResetState(t *testing.T) {
 	}
 }
 
+func TestEmptyVariablesRawString(t *testing.T) {
+	stmt := new(Stmt)
+	stmt.Buf = []rune("select ")
```
Comment (kenshaw) [inline stmt/stmt_test.go:259]: A unit test should not be setting this manually. Use `Append`, which is what's used internally on the call back from `rline`.

File: stmt/stmt_test.go [hunk 3]
```diff
@@ -254,6 +254,29 @@ func TestNextResetState(t *testing.T) {
...
+	stmt.Prefix = "SELECT"
+	v := &Var{
+		I:    7,
+		End:  9,
+		Name: "a",
+		Len:  0,
+	}
+	stmt.Vars = append(stmt.Vars, v)
+
+	if expected, got := "select ", stmt.RawString(); expected != got {
```
Comment (kenshaw) [inline stmt/stmt_test.go:270]: Please change `expected` -> `exp`.

**Comment by xwjdsh** [comment]
I udpated.

**Comment by kenshaw** [comment]
Added.

**Pull request closed without merging**